app/src/main/java/io/actifit/fitnesstracker/actifitfitnesstracker/MainActivity.java

```diff
@@ -1293,10 +1344,71 @@ public void onClick(DialogInterface dialog, int which) {
+                        @Override
+                        public void onClick(View v) {
+                            String url = "https://x.com/intent/like?tweet_id=1797370316022272474";
+                            Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse(url));
+                            startActivity(intent);
+                        }
+                    });
+
+                    ImageView retweet = dialogLayout.findViewById(R.id.retweet);
+                    retweet.setOnClickListener(new OnClickListener(){
+                        @Override
+                        public void onClick(View v) {
+                            String url = "https://x.com/intent/retweet?tweet_id=1797370316022272474";
+                            Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse(url));
+                            startActivity(intent);
+                        }
+                    });
+
+                    ImageView follow = dialogLayout.findViewById(R.id.follow);
+                    follow.setOnClickListener(new OnClickListener(){
+                        @Override
+                        public void onClick(View v) {
+                            String url = "https://x.com/intent/follow?screen_name=Actifit_fitness";
+                            Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse(url));
+                            startActivity(intent);
+                        }
+                    });
+
+                    ImageView reply = dialogLayout.findViewById(R.id.reply);
+                    reply.setOnClickListener(new OnClickListener(){
+                        @Override
+                        public void onClick(View v) {
+                            String url = "https://x.com/intent/tweet?in_reply_to=1797370316022272474";
+                            Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse(url));
+                            startActivity(intent);
+                        }
+                    });
+
+
+                    builder.setPositiveButton("Close", null);
+                    AlertDialog dialog = builder.create();
+                    dialog.show();
+                }
+            });
+
+            freeRewardButton.setOnClickListener(innerView -> showRewardedVideo(innerView, 1,R.id.daily_free_reward,R.id.daily_5k_reward,R.id.daily_7k_reward,R.id.daily_10k_reward));
+            fivekRewardButton.setOnClickListener(innerView -> showRewardedVideo(innerView, 2,R.id.daily_free_reward,R.id.daily_5k_reward,R.id.daily_7k_reward,R.id.daily_10k_reward));
+            sevenkRewardButton.setOnClickListener(innerView -> showRewardedVideo(innerView, 3,R.id.daily_free_reward,R.id.daily_5k_reward,R.id.daily_7k_reward,R.id.daily_10k_reward));
+            tenkRewardButton.setOnClickListener(innerView -> showRewardedVideo(innerView, 4,R.id.daily_free_reward,R.id.daily_5k_reward,R.id.daily_7k_reward,R.id.daily_10k_reward));

```

Comment:
@AliHajjAli1 why did you change the showRewardedVideo function signature and add the 3 params, those params don't change value normally and can be used as constants, as they were. Can you clarify the need for the change?

Reply:
I added those 3 params as the constants, R.id.daily_free_reward,..., were causing an error when using them directly in the method. They were shown as undefined so I put the as params in the method and it works.